File: App_Code/UserPassword.vb
﻿Imports System.Security.Cryptography

Module UserPassword

    Function GenerateSalt() As Byte()
        Dim salt(8) As Byte

        Dim rng As New RNGCryptoServiceProvider()
        rng.GetNonZeroBytes(salt)

        Return salt
    End Function

    Function EncodePassword(ByVal password As String, ByVal salt As Byte()) As Byte()

        Dim fixedSalt(8) As Byte

        fixedSalt = Convert.FromBase64String("+eroclis/")
        password = Convert.ToBase64String(salt) + password + Convert.ToBase64String(fixedSalt)

3:      Dim passwordBytes() As Byte = System.Text.Encoding.UTF8.GetBytes(password)
        Dim hash As SHA1 = SHA1CryptoServiceProvider.Create()

        For i As Integer = 0 To 999

            Dim multiHashPassword() As Byte = hash.ComputeHash(passwordBytes)
            passwordBytes = multiHashPassword

        Next

        Return passwordBytes
    End Function

End Module

File: App_Code/UserPassword.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Security.Cryptography;

/// <summary>
/// Summary description for UserPassword
/// </summary>
public class UserPassword
{
    public UserPassword()
    {
        
    }

    public Byte[] GenerateSalt()
    {
        Byte[] salt = new Byte[8];
        RNGCryptoServiceProvider rng = new RNGCryptoServiceProvider();
        rng.GetNonZeroBytes(salt);
        return salt;
    }

    public Byte[] EncodePassword(String password, Byte[] salt)
    {
        Byte[] fixedSalt = new Byte[8];

        fixedSalt = Convert.FromBase64String("+eroclis/");
        password = Convert.ToBase64String(salt) + password + Convert.ToBase64String(fixedSalt);

        Byte[] passwordBytes = System.Text.Encoding.UTF8.GetBytes(password);

        SHA1 hash = SHA1CryptoServiceProvider.Create();

            for (int i = 0; i < 1000; i++)
        {
            Byte[] multiHashPassword = hash.ComputeHash(passwordBytes);
            passwordBytes = multiHashPassword;
        }

        return passwordBytes;
    }
}